services/grpc/src/main/java/mil/nga/giat/geowave/service/grpc/GeoWaveGrpcServer.java

```diff
@@ -0,0 +1,333 @@
+					request.getBaseParams().getAdapterId().toByteArray());
+			final ByteArrayId indexId = new ByteArrayId(
+					request.getBaseParams().getIndexId().toByteArray());
+
+			// first check to make sure the data store exists
+			if (!storeLoader.loadFromConfig(ConfigOptions.getDefaultPropertyFile())) {
+				throw new ParameterException(
+						"Cannot find store name: " + storeLoader.getStoreName());
+			}
+
+			DataStore dataStore = null;
+			AdapterStore adapterStore = null;
+			GeotoolsFeatureDataAdapter adapter = null;
+
+			// get a handle to the data store and also the adapter store
+			dataStore = storeLoader.createDataStore();
+			adapterStore = storeLoader.createAdapterStore();
+
+			if (adapterId != null) {
+				adapter = (GeotoolsFeatureDataAdapter) adapterStore.getAdapter(adapterId);
+			}
+
+			try (final CloseableIterator<SimpleFeature> iterator = dataStore.query(
+					new QueryOptions(
+							adapterId,
+							indexId),
+					CQLQuery.createOptimalQuery(
+							cql,
+							adapter,
+							null))) {
+
+				while (iterator.hasNext()) {
+					SimpleFeature simpleFeature = iterator.next();
+					SimpleFeatureType type = simpleFeature.getType();
+					Feature.Builder b = Feature.newBuilder();
+					for (int i = 0; i < type.getAttributeDescriptors().size(); i++) {
+						b.putAttributes(
+								type.getAttributeDescriptors().get(
+										i).getLocalName(),
+								simpleFeature.getAttribute(i) == null ? "" : simpleFeature.getAttribute(
+										i).toString());
+					}
+					Feature f = b.build();
+					responseObserver.onNext(f);
+				}
+				responseObserver.onCompleted();
+			}
+			catch (final Exception e) {
+				LOGGER.error("Exception encountered:", e.getStackTrace().toString());
+			}
+		}
+
+		public void spatialQuery(
+				SpatialQueryParameters request,
+				StreamObserver<Feature> responseObserver ) {
+
+			final String storeName = request.getBaseParams().getStoreName();
+			final StoreLoader storeLoader = new StoreLoader(
+					storeName);
+
+			final ByteArrayId adapterId = new ByteArrayId(
+					request.getBaseParams().getAdapterId().toByteArray());
+			final ByteArrayId indexId = new ByteArrayId(
+					request.getBaseParams().getIndexId().toByteArray());
+
+			// first check to make sure the data store exists
+			if (!storeLoader.loadFromConfig(ConfigOptions.getDefaultPropertyFile())) {
+				throw new ParameterException(
+						"Cannot find store name: " + storeLoader.getStoreName());
+			}
+
+			DataStore dataStore = null;
+
+			// get a handle to the data store and also the adapter store
+			dataStore = storeLoader.createDataStore();
+
+			final String geomDefinition = request.getGeometry();
+			Geometry queryGeom = null;
+			try {
+				queryGeom = new WKTReader(
+						JTSFactoryFinder.getGeometryFactory()).read(geomDefinition);
+			}
+			catch (final Exception e) {
+				LOGGER.error("Exception encountered:", e.getStackTrace().toString());
+			}
+
+			final QueryOptions options = new QueryOptions(
```

Comment:
Yes if the string is empty I set the variable to null actually that way the type is explicit instead of accidentally passing in empty string like we discussed.

Reply:
oh, missed it on line 205-208